ISISComputingGroup/EPICS-ioc
# Ticket_8284_Allow_certain_values_to_be_editable_without_restarting_IOC

### Description of work

Added a MACRO for the McLennan IOC that allows a user to be able to set the ASG group of the MOTOR record. The ASG group controls what ASLs (either 1 or 0) are allowed to do I.E (1, read) allows pvs with ASL 1 to be read etc. This concept was used here to allow user to set the JVEL, HLM and LLM without having to change them in the IOC macro screen and thus have to restart the IOC. If they set the MOTOR_ASG to DEFAULT and then start/restart the IOC it means the access control is set to read & write.

- https://github.com/ISISComputingGroup/IBEX/issues/8284

### To test

- Check the code's quality
- Check tests are written correctly and pass
- Set (or don't set) MOTOR_ASG manually in your globals.txt and check functionality by spinning up McLennan IOC using `C:\Instrument\Apps\EPICS\support\IocTestFramework\master>run_all_tests.bat -t mclennan -a`. Remember to let tests run through before `ctrl-c` otherwise you have to find the procserve process in task manager details and end that before you can start up the IOC again 

### Acceptance criteria

- [ ] A security group is created that allows for the JVEL, HLM, and LLM to be set or not based on a specific macro which defaults to not allowing changes
- [ ] There are tests both with and without that macro set to ensure the beaviour is as expected

---

#### Code Review

    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)

### Functional Tests

- IOC responds correctly in:

### Final steps
